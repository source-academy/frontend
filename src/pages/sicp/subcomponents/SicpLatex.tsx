import * as React from 'react';
import Latex from 'react-latex-next';

type SicpLatexProps = {
  math: string;
};

const SicpLatex: React.FC<SicpLatexProps> = props => {
  return <Latex>{props.math}</Latex>;
};

export default SicpLatex;
