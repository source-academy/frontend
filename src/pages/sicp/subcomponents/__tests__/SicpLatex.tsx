import { mount } from 'enzyme';

import SicpLatex from '../SicpLatex';

describe('Sicp latex renders', () => {
  test('correctly block', () => {
    const props = {
      math: '1+1',
      inline: false
    };

    const tree = mount(<SicpLatex {...props} />);
    expect(tree.debug()).toMatchSnapshot();
  });

  test('correctly inline', () => {
    const props = {
      math: '1+1',
      inline: true
    };

    const tree = mount(<SicpLatex {...props} />);
    expect(tree.debug()).toMatchSnapshot();
  });
});
