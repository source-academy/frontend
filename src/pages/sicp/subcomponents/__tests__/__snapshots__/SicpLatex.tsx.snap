// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sicp latex renders correctly block 1`] = `
"<SicpLatex math=\\"1+1\\" inline={false}>
  <Latex delimiters={{...}} strict={false}>
    <span className=\\"__Latex__\\" dangerouslySetInnerHTML={{...}} />
  </Latex>
</SicpLatex>"
`;

exports[`Sicp latex renders correctly inline 1`] = `
"<SicpLatex math=\\"1+1\\" inline={true}>
  <Latex delimiters={{...}} strict={false}>
    <span className=\\"__Latex__\\" dangerouslySetInnerHTML={{...}} />
  </Latex>
</SicpLatex>"
`;
